impeller/geometry/geometry_unittests.cc

```diff
@@ -2267,13 +2267,40 @@ TEST(GeometryTest, PathPolylineDuplicatesAreRemovedForSameContour) {
   ASSERT_EQ(polyline.contours[1].start_index, 4u);
   ASSERT_FALSE(polyline.contours[1].is_closed);
   ASSERT_EQ(polyline.points->size(), 7u);
-  ASSERT_EQ((*polyline.points)[0], Point(50, 50));
-  ASSERT_EQ((*polyline.points)[1], Point(100, 50));
-  ASSERT_EQ((*polyline.points)[2], Point(100, 100));
-  ASSERT_EQ((*polyline.points)[3], Point(50, 50));
-  ASSERT_EQ((*polyline.points)[4], Point(50, 50));
-  ASSERT_EQ((*polyline.points)[5], Point(0, 50));
-  ASSERT_EQ((*polyline.points)[6], Point(0, 100));
+  ASSERT_EQ(polyline.GetPoint(0), Point(50, 50));
+  ASSERT_EQ(polyline.GetPoint(1), Point(100, 50));
+  ASSERT_EQ(polyline.GetPoint(2), Point(100, 100));
+  ASSERT_EQ(polyline.GetPoint(3), Point(50, 50));
+  ASSERT_EQ(polyline.GetPoint(4), Point(50, 50));
+  ASSERT_EQ(polyline.GetPoint(5), Point(0, 50));
+  ASSERT_EQ(polyline.GetPoint(6), Point(0, 100));
+}
+
+TEST(GeometryTest, PolylineBufferReuse) {
+  auto point_buffer = std::make_unique<std::vector<Point>>();
+  auto point_buffer_address = reinterpret_cast<uintptr_t>(point_buffer.get());
+  Path::Polyline polyline =
+      PathBuilder{}
+          .MoveTo({50, 50})
+          .LineTo({100, 100})
+          .TakePath()
+          .CreatePolyline(
+              1.0f, std::move(point_buffer),
+              [point_buffer_address](
+                  Path::Polyline::PointBufferPtr point_buffer) {
+                ASSERT_EQ(point_buffer->size(), 0u);
+                ASSERT_EQ(point_buffer_address,
+                          reinterpret_cast<uintptr_t>(point_buffer.get()));
+              });
+}
+
+TEST(GeometryTest, PolylineFailsWithNullptrBuffer) {
```

Comment:
Thanks for adding these.  A nullptr test for the reclaim callback would be nice too.